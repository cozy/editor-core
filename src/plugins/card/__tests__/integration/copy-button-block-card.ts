import { _getCopyButtonTestSuite } from '../../../../../src/__tests__/integration/copy-button/_getCopyButtonTestSuite';
import type { default as WebDriverPage } from '@atlaskit/webdriver-runner/wd-wrapper';
import * as blockCardAdf from '../../../../__tests__/integration/card/_fixtures_/block-card.adf.json';
import {
  waitForBlockCardSelection,
  waitForDatasourceSelection,
} from '@atlaskit/media-integration-test-helpers';
import * as datasourceAdf from '../../../../__tests__/integration/card/_fixtures_/block-card-datasource.adf.json';

_getCopyButtonTestSuite({
  nodeName: 'Block card',
  editorOptions: {
    smartLinks: {
      allowBlockCards: true,
    },
    defaultValue: JSON.stringify(blockCardAdf),
  },
  nodeSelector: '.card',
  customBeforeEach: async (page: WebDriverPage): Promise<void> => {
    await waitForBlockCardSelection(page);
  },
});

_getCopyButtonTestSuite({
  nodeName: 'Datasource block card',
  editorOptions: {
    smartLinks: {
      allowBlockCards: true,
    },
    defaultValue: JSON.stringify(datasourceAdf),
  },
  nodeSelector: '.card',
  customBeforeEach: async (page: WebDriverPage): Promise<void> => {
    await waitForDatasourceSelection(page);
  },
});
