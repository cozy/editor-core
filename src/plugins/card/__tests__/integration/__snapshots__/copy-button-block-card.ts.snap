// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copy block with floating toolbar copy button 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": " ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "data": Object {
          "@context": "https://www.w3.org/ns/activitystreams",
          "@type": "Document",
          "generator": Object {
            "icon": "https://wac-cdn.atlassian.com/assets/img/favicons/atlassian/favicon.png",
          },
          "name": "Welcome to Atlassian!",
          "url": "http://www.atlassian.com",
        },
        "datasource": null,
        "layout": null,
        "url": null,
        "width": null,
      },
      "type": "blockCard",
    },
    Object {
      "attrs": Object {
        "data": Object {
          "@context": "https://www.w3.org/ns/activitystreams",
          "@type": "Document",
          "generator": Object {
            "icon": "https://wac-cdn.atlassian.com/assets/img/favicons/atlassian/favicon.png",
          },
          "name": "Welcome to Atlassian!",
          "url": "http://www.atlassian.com",
        },
        "datasource": null,
        "layout": null,
        "url": null,
        "width": null,
      },
      "type": "blockCard",
    },
  ],
  "type": "doc",
}
`;
